vortex-serde/src/layouts/read/recordbatchreader.rs

```diff
@@ -0,0 +1,60 @@
+use std::sync::Arc;
+
+use arrow_array::{RecordBatch, RecordBatchReader};
+use arrow_schema::{ArrowError, SchemaRef};
+use futures::StreamExt;
+use vortex::arrow::infer_schema;
+use vortex::Array;
+use vortex_error::{VortexError, VortexResult};
+
+use super::LayoutBatchStream;
+use crate::io::VortexReadAt;
+
+fn vortex_to_arrow_error(error: VortexError) -> ArrowError {
+    ArrowError::ExternalError(Box::new(error))
+}
+
+fn vortex_to_arrow(result: VortexResult<Array>) -> Result<RecordBatch, ArrowError> {
+    result
+        .and_then(RecordBatch::try_from)
+        .map_err(vortex_to_arrow_error)
+}
+
+pub struct VortexRecordBatchReader<R: VortexReadAt + Unpin + Send + 'static> {
+    stream: LayoutBatchStream<R>,
+    arrow_schema: SchemaRef,
+    runtime: tokio::runtime::Runtime,
```

Comment:
👀

Reply:
this is sus, tagging @robert3005 because I don't have an immediately better suggestion

Reply:
I think it makes sense to have the runtime as a static thing, its not really bound to the lifetime of the reader, and I suspect you might run into weird issues if you have multiple runtimes.

Reply:
I recently ran into [this](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#current-thread-scheduler) note in Tokio's docs, the way I read it all readers spawned from the same thread share underlying resources, which essentially makes the IO/serialization serialized between them, which is really bad.

Reply:
Agree with @AdamGS we should have a lazy_static in the library with the runtime that all python functions can use

Reply:
Done.